README.md

```diff
@@ -12,109 +12,831 @@ The repository is a collection of open-source implementation of a variety of alg
+5. [`InsertionSort`](./sort/insertionsort.go#L3): No description provided.
+6. [`Mergesort`](./sort/mergesort.go#L35): Mergesort Perform mergesort on a slice of ints
+7. [`Pigeonhole`](./sort/pigeonholesort.go#L12):  Pigeonhole sorts a slice using pigeonhole sorting algorithm.
+8. [`QuickSort`](./sort/quicksort.go#L37):  QuickSort Sorts the entire array
+9. [`QuickSortRange`](./sort/quicksort.go#L24):  QuickSortRange Sorts the specified range within the array
+10. [`RadixSort`](./sort/radixsort.go#L35): No description provided.
+11. [`SelectionSort`](./sort/selectionsort.go#L3): No description provided.
+12. [`ShellSort`](./sort/shellsort.go#L3): No description provided.
+13. [`SimpleSort`](./sort/simplesort.go#L11): No description provided.
+
+---
+##### Types
+
+1. [`Int`](#L0): 
+
+	Methods:
+	1. [`More`](./sort/heapsort.go#L114): No description provided.
+2. [`MaxHeap`](./sort/heapsort.go#L3): No description provided.
+
+
+---
+</details><details>
+	<summary> <strong> stack </strong> </summary>	
+
+---
+
+##### Types
+
+1. [`Node`](./structure/stack/stacklinkedlist.go#L13): No description provided.
+
+2. [`SList`](./structure/stack/stacklinkedlistwithlist.go#L18): No description provided.
+
+3. [`Stack`](./structure/stack/stacklinkedlist.go#L19): No description provided.
+
+
+---
+</details><details>
+	<summary> <strong> transposition </strong> </summary>	
+
+---
+
+##### Functions:
+
+1. [`Decrypt`](./cipher/transposition/transposition.go#L82): No description provided.
+2. [`Encrypt`](./cipher/transposition/transposition.go#L54): No description provided.
+
+---
+##### Types
+
+1. [`KeyMissingError`](./cipher/transposition/transposition.go#L16): No description provided.
+
+2. [`NoTextToEncryptError`](./cipher/transposition/transposition.go#L15): No description provided.
+
+
+---
+</details><details>
+	<summary> <strong> trie </strong> </summary>	
+
+---
+
+#####  Package trie provides Trie data structures in golang.  Wikipedia: https://en.wikipedia.org/wiki/Trie
+
+---
+##### Functions:
+
+1. [`NewNode`](./structure/trie/trie.go#L14):  NewNode creates a new Trie node with initialized children map.
+
+---
+##### Types
+
+1. [`Node`](./structure/trie/trie.go#L7): No description provided.
+
+
+---
+</details><details>
+	<summary> <strong> xor </strong> </summary>	
+
+---
+
+#####  Package xor is an encryption algorithm that operates the exclusive disjunction(XOR) ref: https://en.wikipedia.org/wiki/XOR_cipher
+
+---
+##### Functions:
+
+1. [`Decrypt`](./cipher/xor/xor.go#L19):  Decrypt decrypts with Xor encryption
+2. [`Encrypt`](./cipher/xor/xor.go#L10):  Encrypt encrypts with Xor encryption after converting each character to byte The returned value might not be readable because there is no guarantee which is within the ASCII range If using other type such as string, []int, or some other types, add the statements for converting the type to []byte.
+
+---
+</details>
+<!--- GODOCMD END --->
```

Comment:
same applies here otherwise it will overwrite everything after `<!--- GODOCMD BEGIN --->`